### PR TITLE
Fixes #2598. Trying to run self-contained application on Ubuntu 22 (Linux-x64), error regarding libdl.so

### LINKED ISSUE #2598: Trying to run self-contained application on Ubuntu 22 (Linux-x64), error regarding libdl.so
**Describe the bug**
I'm getting this error when attempting to run an application I built with .NET 7, Terminal.Gui 1.10.1 targeting Ubuntu 22:
```
Unhandled exception. System.Exception: Curses failed to initialize, the exception is: Unable to load shared library 'libdl.so' or one of its dependencies. In order to help diagnose loading problems, consider using a tool like strace. If you're using glibc, consider setting the LD_DEBUG environment variable.
   at Terminal.Gui.CursesDriver.Init(Action)
   at Terminal.Gui.Application.InternalInit(Func`1, ConsoleDriver , IMainLoopDriver , Boolean )
   at Terminal.Gui.Application.Init(ConsoleDriver , IMainLoopDriver )
   at Program.<Main>$(String[])
Aborted
```
After doing some research, I found this:
https://stackoverflow.com/a/75855054

I also know that v2 is being worked on, so I tried searching on Discussions and Issues for anything about libdl.so but didn't find anything so my apologies if this is already a known issue.

**To Reproduce**
1. Create a simple application using .NET 7 and Terminal.Gui 1.10.1
2. Publish as a single file, self-contained application targeting `linux-x64`
3. chmod +x, attempt to run the application on Ubuntu 22

**Expected behavior**
The application should run, displaying the simple application

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.2 LTS

### PR DESCRIPTION
**Pull request by BDisp**
## Fixes

- Fixes #2598

## Proposed Changes/Todos

- [x] Added NET8_0

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [X] I have checked my code and corrected any poor grammar or misspellings
- [X] I conducted basic QA to assure all features are working

### CODE REVIEW

File: Terminal.Gui/ConsoleDrivers/CursesDriver/UnmanagedLibrary.cs [hunk 1]
```diff
@@ -256,7 +256,7 @@ static class Mono {
 		/// to avoid the dependency on libc-dev Linux.
 		/// </summary>
 		static class CoreCLR {
-#if NET7_0
+#if NET7_0 || NET8_0
```
Comment (dodexahedron) [inline Terminal.Gui/ConsoleDrivers/CursesDriver/UnmanagedLibrary.cs:259]: I think you want `NET7_0_OR_GREATER` instead.

Although conditional compilation only matters if the consumer builds it from source.

**Review by dodexahedron — CHANGES_REQUESTED** [review]
Suggested to use the symbol made for this scenario

File: Terminal.Gui/ConsoleDrivers/CursesDriver/UnmanagedLibrary.cs (hunk 1 above)
Comment (dodexahedron) [reply Terminal.Gui/ConsoleDrivers/CursesDriver/UnmanagedLibrary.cs:259]: Oh hey. Also...

Even though it's v1, this is probably something that should be updated at some point to use LibraryImport, and conditional compilation is nice to be more visible, such as including or excluding things at the project level, or similar.

But also.... Several things that class does are in netstandard2.0 and thus can just use the built-in methods so this doesn't remain a maintenance burden, like the IsWindows and such. 🤷‍♂️ 

And parts that aren't supported in netstandard2.0 but exist in more recent .net can be polyfilled by various source generators out there on nuget, and then platform-specific parts can use more standard facilities from that. The Meziantou.Polyfill is excellent for that and only emits what's actually used, at compile-time.

Comment (BDisp) [reply Terminal.Gui/ConsoleDrivers/CursesDriver/UnmanagedLibrary.cs:259]: `LibraryImport` is not supported in all project targets, so I think it is not a good alternative.

image

Comment (BDisp) [reply Terminal.Gui/ConsoleDrivers/CursesDriver/UnmanagedLibrary.cs:259]: I will use `NET6_0_OR_GREATER` to support `net6.0` which is still currently supported (LTS).

**Review by tig — APPROVED** [review]

**Pull request merged**